app/src/components/common/PasswordValidation.tsx

```diff
@@ -1,19 +1,23 @@
-import {List, ListItem, ListSubheader, Stack} from '@mui/material';
+import {List, ListItem, ListSubheader} from '@mui/material';
 import RemoveRoundedIcon from '@mui/icons-material/RemoveRounded';
 import CheckRoundedIcon from '@mui/icons-material/CheckRounded';
 import {useEffect, useState} from 'react';
 import pwValidate from '../../utils/PasswordValidationSchema';
 
 interface PasswordProps {
   password: string;
+  confirmPassword?: string;
```

Comment:
I don't think `confirmPassword` should be an optional parameter. I think it makes sense to update the `SignUp` page to include password matching validation, and remove the password validation from the `SignIn` in page.

`SignIn` pages usually don't have validation since the user is not creating a password, and that might make guessing passwords a bit easier

Reply:
Great point on removing validation from the sign in page. Maybe just an error message for making it required?

Reply:
I was thinking that once we remove the password validation from the sign in page, that we could require all users of the `PasswordProps` interface / `PasswordValidation` component to pass in both a `password` and `confirmPassword`. 

That would force us to always apply the same matching validation each time a new password was specified (for example, we are currently missing the password matching validation on the 'new password' form). 

It would also simplify the `PasswordValidation` logic a bit, since you could remove the `confirmPassword !== undefined` check. 

All that stuff is optional though - I think removing validation from the sign in page is a high priority. Adding password matching validation to the sign up form is a low priority, and polishing up the `PasswordProps` interface is also a low priority.

Reply:
`const validationSchema = object({
  password: string()
    .required('password is required')
    .min(8, 'password must be at least 8 characters')
    .matches(
      /^(?=.*[a-z])/,
      'password must contain at least one lowercase character',
    )
    .matches(/^(?=.*[0-9])/, 'password must contain at least one number')
    .matches(
      /^(?=.*[A-Z])/,
      'password must contain at least one uppercase character',
    )
    .matches(
      /^(?=.*[!@#%&])/,
      'password must contain at least one special character',
    ),
  confirmPassword: string()
    .required('confirm password is required')
    .oneOf([ref('password'), null], 'passwords must match'),
});`

Currently, this validation schema in `src/components/authentication/NewPasswordForm.tsx` handles the password matching validation. 

Great idea on consolidating this validation schema (password matching validation) with the `PasswordValidation` schema. I'll look into it.

Reply:
Hey @rochun, thanks for volunteering! It might be a good idea to create an issue that describes this change, so we don't forget about it.